soh/soh/Enhancements/TimeSavers/CrawlSpeed.cpp

```diff
@@ -77,9 +77,9 @@ void CrawlSpeed_Register() {
     COND_VB_SHOULD(VB_CRAWL_SPEED_EXIT_CS, shouldRegister, {
         Player* player = GET_PLAYER(gPlayState);
         Camera* csCam = va_arg(args, Camera*);
-        s16 csId = va_arg(args, s16);
-        s16 actionParameters = va_arg(args, s16);
-        s16 initTimer = va_arg(args, s16);
+        s16 csId = (s16)va_arg(args, int);
+        s16 actionParameters = (s16)va_arg(args, int);
+        s16 initTimer = (s16)va_arg(args, int);
```

Comment:
great catch here handling the va_arg promotion https://en.cppreference.com/w/cpp/language/variadic_arguments, https://en.cppreference.com/w/cpp/language/implicit_conversion#Integral_promotion

tiny cleanup suggestion since we're in a cpp file

```suggestion
        s16 csId = static_cast<s16>(va_arg(args, int32_t));
        s16 actionParameters = static_cast<s16>(va_arg(args, int32_t));
        s16 initTimer = static_cast<s16>(va_arg(args, int32_t));
```

Reply:
`int32_t` runs into same issue on platforms where `int` isn't 32 bits

Reply:
Yeah, generally we stick to static_cast in all cpp files. There are still some lingering c-style casts in them, but it's just best to utilize the higher technology (also makes it easier to find things like this in the future should someone decide to try to actually resolve it rather than just patch over it like we're doing for now).